modules/hydrodyn/src/Waves.f90

```diff
@@ -1939,13 +1939,13 @@ CASE ( 1, 10 )          ! Plane progressive (regular) wave; the wave spectrum is
          ELSE
             ! The elevation of the point defined by WaveKinzi(J) must lie between the seabed and the mean sea level (inclusive)
 
-            InitOut%WaveDynP(:,J  ) = WaveDynP0B(:,K)
-            InitOut%WaveVel (:,J,1) = WaveVel0Hxi(:,K)
-            InitOut%WaveVel (:,J,2) = WaveVel0Hyi(:,K)
-            InitOut%WaveVel (:,J,3) = WaveVel0V(:,K)
-            InitOut%WaveAcc (:,J,1) = WaveAcc0Hxi(:,K)
-            InitOut%WaveAcc (:,J,2) = WaveAcc0Hyi(:,K)
-            InitOut%WaveAcc (:,J,3) = WaveAcc0V(:,K)
+            InitOut%WaveDynP(0:InitOut%NStepWave-1,J  ) = WaveDynP0B( 0:InitOut%NStepWave-1,K)
```

Comment:
The first index on the arrays on the left runs from `0:InitOut%NStepWave` instead of `0:InitOut%NStepWave-1` like the right